### PR TITLE
install gnupg in the base ubuntu:18.04 image

### LINKED ISSUE #727: docker core fails to build
https://app.travis-ci.com/github/SyneRBI/SIRF-SuperBuild/jobs/572981957#L801-L804
```
Step 6/33 : RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
 ---> Running in d325922b9b53
E: gnupg, gnupg2 and gnupg1 do not seem to be installed, but one of them is required for this operation
The command '/bin/sh -c apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub' returned a non-zero code: 255
```
The GPU-core does build on Travis (amazing!).

We shouldn't need to have to do NVidia stuff on the non-GPU docker really, but if it's easier, then fine for me.

### PR DESCRIPTION
**Pull request by paskino**
fixes #727

### CODE REVIEW

File: docker/update_nvidia_keys.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+
+if [ "${PYTHON_EXECUTABLE}" = "ubuntu:18.04" ]; then
```
[the remark this answers is not in the document]
Comment (paskino) [reply docker/update_nvidia_keys.sh:3]: no!

**Comment by paskino** [comment]
@KrisThielemans how do I start the travis build?

**Comment by paskino** [comment]
I triggered a build but got 
> **Oh no!** You tried to trigger a build for SyneRBI/SIRF-SuperBuild but the request was rejected.

File: docker/update_nvidia_keys.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    apt-get update -y && apt-get install -yq gnupg wget &&\
+    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+
+} || {
+
+    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+
+}
```
Comment (KrisThielemans) [inline docker/update_nvidia_keys.sh:13]: why the somewhat complicated if? Does `apt install gnupg wget` fail on the cuda image?

If it does, I think the following is a lot shorter/clearer
```suggestion
    apt update -y
    apt install -yq gnupg wget || true
    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
```

**Comment by KrisThielemans** [comment]
Looks like you cannot trigger a build, but I can. No idea how to change that. Sorry.

**Comment by paskino** [comment]
@KrisThielemans if you can trigger a build of this, I could then merge this.

File: docker/update_nvidia_keys.sh (hunk 2 above)
Comment (KrisThielemans) [reply docker/update_nvidia_keys.sh]: I guess that wasn't what I was asking. My question was if the `apt install -yq gnupg` ever fails. If not, why do we need the `|| true`?
Anyway, I don't mind having it.

File: docker/update_nvidia_keys.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#! /bin/bash
+
+# similar to try/catch
+# this line is required for the ubuntu:18.04 base image
+apt-get update -y && apt-get install -yq gnupg wget || true
+    
+
+# https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
+apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+apt-get install software-properties-common -yq
```
Comment (KrisThielemans) [inline docker/update_nvidia_keys.sh:10]: why is this here? I know we need it (see #734), but it seems to have nothing to do with nvidia keys

**Comment by paskino** [comment]
Installs `add-apt-repository` to address

https://app.travis-ci.com/github/SyneRBI/SIRF-SuperBuild/jobs/576061608#L1741-L1742

File: docker/update_nvidia_keys.sh (hunk 3 above)
Comment (paskino) [reply docker/update_nvidia_keys.sh:10]: I didn't know. OK, I merged #734 and will merge it in here.

**Review by KrisThielemans — APPROVED** [review]

**Comment by KrisThielemans** [comment]
let's squash-merge this one

**Comment by KrisThielemans** [comment]
> @KrisThielemans if you can trigger a build of this, I could then merge this.

I don't think I can trigger a build of a branch on your fork. Let's just squash-merge to master and then I'll trigger.

I'm assuming you have run this locally?

File: docker/update_nvidia_keys.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
+# similar to try/catch
+# this line is required for the ubuntu:18.04 base image
+apt-get update -y && apt-get install -yq gnupg wget || true
```
Comment (KrisThielemans) [inline docker/update_nvidia_keys.sh:5]: Codacy isn't happy about this sequence of && and ||:
> Error-prone. Note that A && B || C is not if-then-else. C may run when A is true.

I agree with that really. let's split it into 2 lines

File: docker/update_nvidia_keys.sh (hunk 2 above)
Comment (KrisThielemans) [reply docker/update_nvidia_keys.sh]: ok. I think I understand. On the GPU image, anything with `apt` fails before adding the keys. sorry.

File: docker/update_nvidia_keys.sh (hunk 4 above)
Comment (paskino) [reply docker/update_nvidia_keys.sh:5]: but the problem is that `apt update` would fail if the keys are not right

Comment (KrisThielemans) [inline docker/update_nvidia_keys.sh:5]:
```suggestion
# these lines are required for the ubuntu:18.04 base image, 
# but we allow them to fail as they can on a GPU image before the keys as added.
apt-get update -y  || true
apt-get install -yq gnupg wget || true
```

**Comment by paskino** [comment]
Yes, I've run locally and it works.

**Pull request merged**